docs/guide/helper-overview.md

```diff
@@ -67,12 +67,16 @@ class ArrayHelper extends BaseArrayHelper
 
 Save your class in a file named `ArrayHelper.php`. The file can be in any directory, for example `@app/components`.
 
-Next, in your application's [entry script](structure-entry-scripts.md), add the following line of code
-after including the `yii.php` file to tell the [Yii class autoloader](concept-autoloading.md) to load your custom
+Next, in your application's [entry script](structure-entry-scripts.md), add the class to the class map
+to tell the [Composer class autoloader](concept-autoloading.md) to load your custom
 class instead of the original helper class from the framework:
 
 ```php
-Yii::$classMap['yii\helpers\ArrayHelper'] = '@app/components/ArrayHelper.php';
+/** @var \Composer\Autoload\ClassLoader $loader */
+$loader = require __DIR__ . '/vendor/autoload.php';
+$loader->addClassMap([
+    'yii\helpers\ArrayHelper' => 'components/ArrayHelper.php',
```

Comment:
Path to file should be absolute.

I'm also against such approach, since it kills some composer autoloader optimizations. For PHP 5.6+ composer uses opcache to store static array with classmap: https://getcomposer.org/doc/articles/autoloader-optimization.md#what-does-it-do- . You will lose this if you modify classmap. Overrides should be defined in `composer.json` - then it will be included in composer classmap and no additional processing will be needed.

Reply:
`composer.json` doesn't support per-class mappings.

Reply:
Yes it does, in multiple ways, for example:

```json
    "autoload": {
        "psr-4": {
          "app\\": ""
        },
        "classmap": ["overrides/yii/helpers/Html.php"],
        "exclude-from-classmap": ["vendor/yiisoft/yii2/helpers/Html.php"]
    },
```

`exclude-from-classmap` section is required to prevent errors with `-o` option used in `dump-autoload`.

Reply:
Oh... I'll check it.